docs/learn/create-a-plugin/create-a-plugin.md

```diff
@@ -0,0 +1,171 @@
+
+There are several types of plugins you can create for your Scully app, depending on when in the build pipeline you need to jump in. Here are the different types of plugins you can create and a brief explanation of each, [pulled directly from Scully's docs](https://scully.io/docs/Reference/plugins/overview/):
+
+- `router` plugins teach Scully how to get the required data to be pre-render pages from the route-params.
+- `render` plugins are used to transform the rendered HTML. After the Angular application renders, the HTML content is passed to a render plugin where it can be further modified.
+- `routeProcess` plugins are plugins that can modify the handled route array, before rendering the routes starts
+- `fileHandler` plugins are used by the `contentFolder` plugin during the render process. The `contentFolder` plugin processes the folders for markdown files or other file type the folders may contain. The render process processes any existing `fileHandler` plugin for any file extension type.
+- `routeDiscoveryDone` plugins are called automatically after all routes have been collected and all `router` plugins have finished.
+- `allDone` plugins are like `routeDiscoveryDone` plugins, except they are called after Scully finishes executing all its processes.
+- Scully has a category of system plugins. Unlike the other plugin categories those plugins don't have a set interface, and do use a symbol for their name.
+
+With these seven types of plugins, you can create a lot of extra functionality in your Scully app. For example, I recently wanted the title from my blog posts (written in Markdown) to be added as the HTML document's title. There are several ways to do this, but one way is to write a custom `render` plugin. This plugin gives you access to the rendered HTML of a page, as well as some data about the route, and allows you to alter it in some way. In my case, I looked for the `title` attribute on the route's data and added that to the rendered HTML.
+
+Another type of plugin that's useful is the `routeDiscoveryDone` plugin. This type of plugin is called after Scully finds all the routes in the app and any `router` plugins are done running. A use case for this plugin is creating an RSS feed from the routes in your Scully application. You can see [an example here](https://github.com/notiz-dev/scully-plugins).
+
+Now that we've covered the types of plugins that you can create and a couple examples of use cases, let's look at how you can create a custom plugin.
+
+## [Plugin Return Values](#plugin-return-values)
+
+All Scully plugins return a `Promise<unknown>`, as illustrated in the interface:
+
+```ts
+interface plugin {
+  (...parameters:any) => Promise<unknown>
+}
+```
+
+For example, a render plugin returns a Promise<string>. A `router` plugin returns an array of `HandledRoute`s wrapped in a `Promise`. Some plugins don't actually return anything, but the return value is still wrapped in a `Promise`, so its return value is `Promise<void>`.
```

Comment:
`render` instead of render.

Reply:
Also... maybe say `Promise<HandledRoute[]>` instead of `an array of HandledRoutes wrapped in a promise`. Or, at the end of the long explanation, add the TypeScript description of the same thing.

Reply:
I made both of those changes @aaronfrost. Thanks for the feedback!